### PR TITLE
wgengine/netstack: don't pass non-subnet traffic to netstack in hybrid mode

### PR DESCRIPTION
**Pull request by bradfitz**
Fixes tailscale/corp#1725

### CODE REVIEW

File: wgengine/netstack/netstack.go [hunk 1]
```diff
@@ -374,6 +374,12 @@ func (ns *Impl) injectOutbound() {
 }
 
 func (ns *Impl) injectInbound(p *packet.Parsed, t *tstun.Wrapper) filter.Response {
+	if ns.onlySubnets && tsaddr.IsTailscaleIP(p.Dst.IP) {
```
Comment (bradfitz) [inline wgengine/netstack/netstack.go:377]: IsTailscaleIP is slightly lazy and potentially incorrect down the road if we start handing out non-CGNAT Tailscale IPs to people.

But it's also very easy and cheap compared to comparing netmaps.

I could put some `atomic.Value` of `func(netaddr.IP) bool` in the `netstack.Impl` type I guess.

Comment (bradfitz) [reply wgengine/netstack/netstack.go:377]: Like we already have in `wgengine/userspace.go`:

```go
        // isLocalAddr reports the whether an IP is assigned to the local                                                                                                               
        // tunnel interface. It's used to reflect local packets                                                                                                                         
        // incorrectly sent to us.                                                                                                                                                      
        isLocalAddr atomic.Value // of func(netaddr.IP)bool                                                                                                                             
```

**Review by crawshaw — APPROVED** [review]

File: wgengine/netstack/netstack.go (hunk 1 above)
Comment (danderson) [reply wgengine/netstack/netstack.go:377]: s'fine, we assume IsTailscaleIP in a ton of places already.

**Review by danderson — APPROVED** [review]

**Comment by bradfitz** [comment]
PTAL; now without the assumptions.

This should actually be faster too, not that IsTailscaleIP is super slow.

**Review by danderson — APPROVED** [review]

**Comment by bradfitz** [comment]
Fixed a doc thing, merging.

**Pull request merged**